release_process.md

```diff
@@ -1,162 +1,550 @@
+- Version format: `X.Y.Z-beta.N`
+- After RC creation, bumped to next minor version with `-beta.0` (unreleased)
+  - Beta previews published by bumping to `-beta.1+`
+
+### Release Branches
+- Format: `release/v{major}.{minor}` (e.g., `release/v1.3`)
+- Created when cutting initial RC for major/minor release
+- Maintained for patch releases
+- Version progression: `rc.1` → `rc.2` → stable → `beta.0` → `rc.1` (for patches)
+
+## Version Flow
+
+```mermaid
+%%{init: {'theme':'base', 'themeVariables': { 'fontSize':'14px'}}}%%
+flowchart LR
+    subgraph main["Main Branch"]
+        direction LR
+        M0["1.3.0-beta.2<br/>📍 release-root/1.3.0-beta.N"] --> M1["1.4.0-beta.0<br/>📍 release-root/1.4.0-beta.N"]
+        M1 --> M2["1.4.0-beta.1<br/>🏷️ v1.4.0-beta.1"]
+        M2 --> M3["1.5.0-beta.0<br/>📍 release-root/1.5.0-beta.N"]
+    end
+
+    subgraph release["Release Branch: release/v1.3"]
+        direction LR
+        R1["1.3.0-rc.1<br/>🏷️ v1.3.0-rc.1"] --> R2["1.3.0<br/>🏷️ v1.3.0"]
+        R2 --> R3["1.3.1-beta.0"]
+        R3 --> R4["1.3.1-rc.1<br/>🏷️ v1.3.1-rc.1"]
+        R4 --> R5["1.3.1<br/>🏷️ v1.3.1"]
+        R5 --> R6["1.3.2-beta.0"]
+    end
+
+    M0 -.->|"create RC<br/>(no breaking changes)"| R1
+```
 
-We create a full release of Lance up to every 2 weeks. In between full releases,
-we make preview releases of the latest features and bug fixes, which are hosted
-on fury.io. This allows us to release frequently and get feedback on new features
-while keeping under the PyPI project size limits.
+**Flow explanation:**
+- **Main branch** at `1.3.0-beta.2` (has release-root tag) → create RC checks for breaking changes → if none, RC is `1.3.0-rc.1`, main bumps to `1.4.0-beta.0` with new release-root tag → `1.4.0-beta.1` (preview, tagged) → create RC for v1.4.0 → main bumps to `1.5.0-beta.0`
+- **Release branch** `release/v1.3` created from main at `1.3.0-beta.2`, starts at `1.3.0-rc.1` (tagged) → `1.3.0` (stable, tagged) → `1.3.1-beta.0` → `1.3.1-rc.1` (tagged) → `1.3.1` (stable, tagged) → `1.3.2-beta.0`
+- **Tags**: 🏷️ = version tag, 📍 = release-root tag (for breaking change detection)
+- **Breaking changes**: If breaking changes detected when cutting RC from `1.3.0-beta.2`, the RC would be `2.0.0-rc.1` and main would bump to `2.1.0-beta.0`
 
-## Overview of Automated Release Process
+**Note**: All commits are linear on their respective branches. `beta.0` = unreleased, `beta.1+` = published previews.
 
-The Lance release process is now automated using `bump-my-version` to eliminate 
-manual version updates. The workflow handles version bumping, breaking change 
-validation, and release creation automatically.
+## Workflows
 
-## Choosing a full versus preview release
+### User-Facing Workflows
 
-There are three conditions that can trigger a full release:
+1. **publish-beta.yml** - Publish beta preview releases from any branch
+2. **create-major-minor-rc.yml** - Create initial RC for new major/minor version
```

Comment:
Could we instead call this `create-release-branch`? I think that's more the action I would think of wanting to do. "create-major-minor-rc" feels more wordy and unclear.